app/main/posts/views/filters/filter-posts.directive.js

```diff
@@ -5,32 +5,70 @@ function FilterPostsDirective() {
     return {
         restrict: 'E',
         scope: {
-            filters: '='
+            filters: '=',
+            onOpen: '&',
+            onClose: '&'
         },
         replace: true,
         controller: FilterPostsController,
         template: require('./filter-posts.html')
     };
 }
 
-FilterPostsController.$inject = ['$scope', '$timeout','ModalService', 'PostFilters', '$state'];
-function FilterPostsController($scope, $timeout, ModalService, PostFilters, $state) {
+FilterPostsController.$inject = ['$scope', 'PostFilters', '$state', '$document', '$element'];
+function FilterPostsController($scope, PostFilters, $state, $document, $element) {
     $scope.searchSavedToggle = false;
-    $scope.qFilter = '';
+    $scope.status = { isopen: false };
     $scope.view = $state.params.view;
+    $scope.hideDropdown = hideDropdown;
+    $scope.showDropdown = showDropdown;
+    $scope.removeQueryFilter = removeQueryFilter;
+    $scope.applyFilters = applyFilters;
     activate();
 
-    $scope.status = {
-        isopen: false
-    };
-
-    $scope.toggleDropdown = function ($event) {
-        $event.preventDefault();
-        $event.stopPropagation();
-        $scope.status.isopen = !$scope.status.isopen;
-    };
     function activate() {
-        // @todo define initial filter values
-        // $scope.$watch('filters', handleFilterChange, true);
+        // Watch all click events on the page
+        $document.on('click', handleDocumentClick);
+
+        $scope.$on('$destroy', () => {
+            $document.off('click', handleDocumentClick);
+        });
+
+        $scope.$watch('status.isopen', (value) => {
+            if (value) {
+                $scope.onOpen();
+            } else {
+                $scope.onClose();
+            }
+        });
+    }
+
+    function applyFilters() {
+        PostFilters.reactiveFilters = true;
+        $scope.status.isopen = false;
+    }
+
+    function removeQueryFilter() {
+        PostFilters.clearFilter('q', '');
+    }
+
+    function showDropdown() {
+        $scope.status.isopen = true;
+    }
+
+    function hideDropdown() {
+        $scope.status.isopen = false;
+    }
+
+    // Close the dropdown for any clicks outside of the filters
+    function handleDocumentClick(evt) {
```

Comment:
Had to do this ourselves, because the uib-dropdown would close when clicking into the q input

Reply:
That's unfortunate, may be add a comment referencing uib-dropdown